core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2252,6 +2252,20 @@ object ZIO extends ZIOCompanionPlatformSpecific {
   )(f: PartialFunction[A, U]): ZIO[R, E, List[U]] =
     ZIO.collectAllParN(n)(in).map(_.collect(f))
 
+  /**
+   * Evaluate each effect in the structure from left to right, collecting the
+   * the successful values and discarding the empty cases. For a parallel version, see `collectSomePar`.
+   */
+  def collectSome[R, E, A, B](in: Iterable[A])(f: A => ZIO[R, Option[E], B]): ZIO[R, E, List[B]] =
```

Comment:
I would be inclined to name this `collect`.

Reply:
Just a heads up - if we change the name to `collect` then `collectPar` and `collectParN` are not going to be defined at the same place. The `collectAll*` functions will come in between.

Reply:
I think that is fine.